### ISSUE TITLE
Add ability to provide a JSON date parser

### ISSUE THREAD

**Issue by mattchidley** [JeringTech/Javascript.NodeJS#136]
serverOnRequestListener appears to be where the request body params is initially parsed as JSON. In an ideal world a developer could provide a date parser or the name of a date parser function by loading the module (similar to how method invocation is done).

Has this been discussed at all?

**Comment by JeremyTCD** [comment]
Hi, do you want to pass a date from C# to Node and parse it there using a custom date parser?

I've created an example for that: https://github.com/JeremyTCD/Example.DateParser.

To use the example:
-  clone
- `yarn install` in `path/to/Example.DateParser/DateParserExample/JavascriptJavascript` directory
- start without debugging

Core code:
```cs
internal class Program
{
    static async Task Main(string[] _)
    {
        // Configure path where StaticNodeJSService will look for modules
        StaticNodeJSService.Configure<NodeJSProcessOptions>(options =>
        {
            options.ProjectPath = Path.Combine(Directory.GetCurrentDirectory(), "Javascript");
        });

        // Dummy module that imports date-format-parse (https://www.npmjs.com/package/date-format-parse)            
        const string dummyModule = @"
const dateFormatParse = require('date-format-parse');

module.exports = (callback, rawDate) => {
    const date = dateFormatParse.parse(rawDate, 'YYYY-MM-DD HH:mm:ss');
    callback(null, date.getFullYear());
}";

        // Invoke module
        int resultYear = await StaticNodeJSService.InvokeFromStringAsync<int>(dummyModule, args: new string[] { "2019-12-10 14:11:12" });

        // Year should be 2019
        Console.WriteLine(resultYear);
    }
}
```

**Comment by mattchidley** [comment]
@JeremyTCD no, that's not the intention.

The rawDate in your example is string, and needs to be parsed a second time to Date before you can work with it as Date. Similarly, if we serialize a DateTime object from C#, it is parsed by the JavaScript as string **unless you provide a custom parser to the json.Parse method**

json.Parse supports receiving a provided custom parser but that functionality is not exposed in serverOnRequestListener... so for every Date property on the JS side we must double-parse before we can work with it as Date

**Comment by JeremyTCD** [comment]
Specifying json.parse revivers is not supported. PRs welcome though.

**Comment by JeremyTCD** [comment]
Closing due to inactivity. Feel free to open a new issue if you have any other questions.
